src/core.jl

```diff
@@ -1571,10 +1559,14 @@ A user can provide their own initial `grad_ys` to compute the derivatives using
 
 see: [Python Docs](https://www.tensorflow.org/versions/master/api_docs/python/tf/gradients)
 """
-function gradients(y, x::AbstractArray, grad_y=nothing)
+gradients(y, x::AbstractArray, dx=nothing) = pyg_gradients(y, x, dx)
+gradients(y, x, grad_y=nothing) = gradients(y, [x], grad_y)[1]
+
+function pyg_gradients(y, x::AbstractArray, grad_y=nothing)
```

Comment:
What does the 'g' in `pyg` stand for? If it's `gradient`, then it seems redundant with the `_gradients` part of the name.

Reply:
'g' stands for Not **g**etting a  clash with `py_gradients` defined in `py.jl`
I think a better name for this 
would be `py_add_gradients` and `c_add_gradients` maybe?

I messed with the names because I was trying to get `c_gradients` to work,  and wanted different overloads
but really that part of the PR could just be backed out and removed if you'ld prefer